### PR TITLE
cmr url injection

### PR DESCRIPTION
**Pull request by ndp-opendap**
This PR adds code to support identifying the CMR query endpoint URL in the bes.conf chain.

### CODE REVIEW

File: modules/ngap_module/NgapApi.cc [hunk 1]
```diff
@@ -79,6 +79,17 @@ namespace ngap {
             "kNumberType"
     };
 
+
+    NgapApi::NgapApi() : d_cmr_endpoint_url(DEFAULT_CMR_ENDPOINT_URL) {
+        bool found;
+        string key_value;
+        TheBESKeys::TheKeys()->get_value(NGAP_CMR_ENDPOINT_URL, key_value, found);
```
Comment (jgallagher59701) [inline modules/ngap_module/NgapApi.cc]: No need to change this code - it works - but there are other functions in TheBESKeys that are easier to use.

File: modules/ngap_module/NgapCache.cc [hunk 2]
```diff
@@ -237,7 +237,7 @@ NgapCache::get_hash(const string &s)
     }
 
 
-    string NgapCache::get_cache_file_name( const string &src,  bool /*mangle*/){
+    string NgapCache::get_cache_file_name( const string &,  bool /*mangle*/){
```
Comment (jgallagher59701) [inline modules/ngap_module/NgapCache.cc]: Throw an exception.

Comment (ndp-opendap) [reply modules/ngap_module/NgapCache.cc]: It already does that

File: modules/ngap_module/NgapApi.cc (hunk 1 above)
Comment (ndp-opendap) [reply modules/ngap_module/NgapApi.cc]: I bam not aware of these. All of the examples of this that I I have found are of this form.

**Pull request merged**